Lib/test/test_descr.py

```diff
@@ -1835,29 +1835,24 @@ def __init__(self, foo):
         object.__init__(A(3))
         self.assertRaises(TypeError, object.__init__, A(3), 5)
 
-    @unittest.expectedFailure
     def test_restored_object_new(self):
         class A(object):
             def __new__(cls, *args, **kwargs):
-                raise AssertionError
-        self.assertRaises(AssertionError, A)
+                raise TypeError('A')
```

Comment:
It is better to use anything other than TypeError, because TypeError can be indirectly raised by many other reasons. AssertionError was used because it is only indirectly raised in `assert`.